disperser/common/blobstore/blob_metadata_store.go

```diff
@@ -96,6 +96,21 @@ func (s *BlobMetadataStore) GetBlobMetadataByStatus(ctx context.Context, status
 	return metadata, nil
 }
 
+// GetBlobMetadataByStatusCount returns the count of all the metadata with the given status
+// Because this function scans the entire index, it should only be used for status with a limited number of items.
+// It should only be used to filter "Processing" status. To support other status, a streaming version should be implemented.
+func (s *BlobMetadataStore) GetBlobMetadataByStatusCount(ctx context.Context, status disperser.BlobStatus) (int32, error) {
```

Comment:
similarly, let's add [a test](https://github.com/Layr-Labs/eigenda/blob/master/disperser/common/blobstore/blob_metadata_store_test.go) for this as well

Reply:
Added method call to existing test